File: backend/app/services/taskServices.py
from typing import List, Optional
from sqlmodel import Session, select
from app.models import Task, TaskCreate, TaskUpdate  # Assurez-vous d'importer les modèles appropriés

def create_task(db: Session, task: TaskCreate, user_id: int) -> Task:
    db_task = Task(**task.model_dump(), user_id=user_id)
    db.add(db_task)
    db.commit()
    db.refresh(db_task)
    return db_task

def get_task(db: Session, task_id: int, user_id: int) -> Optional[Task]:
    task = db.exec(select(Task).where(Task.user_id == user_id, Task.id == task_id)).first()
    return task

def get_tasks(db: Session, user_id: int, skip: int = 0, limit: int = 10) -> List[Task]:
    tasks = db.exec(select(Task).where(Task.user_id == user_id).offset(skip).limit(limit)).all()
    return tasks

def update_task(db: Session, task_id: int, task_update: TaskUpdate) -> Optional[Task]:
    db_task = db.get(Task, task_id)
    if db_task:
        task_data = task_update.model_dump(exclude_unset=True)
        for key, value in task_data.items():
            setattr(db_task, key, value)
        db.commit()
        db.refresh(db_task)
    return db_task

def delete_task(db: Session, task_id: int) -> bool:
    db_task = db.get(Task, task_id)
    if db_task:
        db.delete(db_task)
        db.commit()
        return True
    return False

File: frontend/src/services/taskServices.ts
import { Task } from "../utils/types/todoify";
import { api } from "./api";

export const tasksService = {
  getTasks: async (token: string): Promise<Task[]> => {
    const response = await api(token).get<Task[]>(`/tasks/?skip=0&limit=100`);
    return response.data;
  },
  getTaskById: async (taskId: number, token: string): Promise<Task | null> => {
    const response = await api(token).get<Task>(`/tasks/${taskId}`);
    return response.data;
  },
  createTask: async (task: Task, token: string): Promise<Task> => {
    const response = await api(token).post<Task>("/tasks/", task);
    return response.data;
  },
  updateTask: async (
    taskId: number,
    task: Task,
    token: string
  ): Promise<Task> => {
    const response = await api(token).put<Task>(`/tasks/${taskId}`, task);
    return response.data;
  },
  deleteTask: async (taskId: number, token: string): Promise<void> => {
    await api(token)
      .delete(`/tasks/${taskId}`)
      .catch((error) => {
        throw Error(error);
      });
  },
};
